packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -328,103 +336,103 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
+
     siblings.forEach(sibling => {
-      if (hasSubtree(sibling) && !this.isActiveItem(sibling['id'])) {
-        activeItemIds.push(sibling['id']);
+      if (hasSubtree(sibling) && !isActiveItem(sibling['id'])) {
+        nextActiveItemsIds.push(sibling['id']);
       }
     });
 
-    if (hasSubtree(treeItemProps) && !this.isActiveItem(id)) {
-      activeItemIds.push(id);
+    if (hasSubtree(treeItemProps) && !isActiveItem(id)) {
+      nextActiveItemsIds.push(id);
     }
 
-    this.setActiveItemIds(e, activeItemIds);
-  };
-
-  setActiveItemIds = (e: React.SyntheticEvent, activeItemIds: string[]) => {
-    _.invoke(this.props, 'onActiveItemIdsChange', e, { ...this.props, activeItemIds });
-
-    this.setState({
-      activeItemIds,
-    });
+    setActiveItemIds(e, nextActiveItemsIds);
   };
 
-  getAllSelectableChildrenId = items => {
+  const getAllSelectableChildrenId = items => {
     return items.reduce((acc, item) => {
       if (item.items) {
-        return [...acc, ...this.getAllSelectableChildrenId(item.items)];
+        return [...acc, ...getAllSelectableChildrenId(item.items)];
       }
       return item.hasOwnProperty('selectable') && !item.selectable ? acc : [...acc, item.id];
     }, []);
   };
 
-  isIndeterminate = (item: TreeItemProps) => {
+  const isIndeterminate = (item: TreeItemProps) => {
     if (!item.selectableParent || !item.items) {
       return false;
     }
 
     const { items } = item;
 
-    const selectableItemIds = this.getAllSelectableChildrenId(items);
+    const selectableItemIds = getAllSelectableChildrenId(items);
 
-    return !this.isAllGroupChecked(items) && selectableItemIds.some(id => this.state.selectedItemIds.indexOf(id) > -1);
+    return !isAllGroupChecked(items) && selectableItemIds.some(id => selectedItemIds.indexOf(id) > -1);
   };
 
-  isAllGroupChecked = (items: ShorthandCollection<TreeItemProps, never>) => {
-    const selectableItemIds = this.getAllSelectableChildrenId(items);
-    return selectableItemIds.every(id => this.state.selectedItemIds.indexOf(id) > -1);
+  const isAllGroupChecked = (items: ShorthandCollection<TreeItemProps, never>) => {
+    const selectableItemIds = getAllSelectableChildrenId(items);
+    return selectableItemIds.every(id => selectedItemIds.indexOf(id) > -1);
   };
 
-  contextValue: TreeRenderContextValue = {
-    onFocusParent: this.onFocusParent,
-    onSiblingsExpand: this.onSiblingsExpand,
-    onFocusFirstChild: this.onFocusFirstChild,
-    onTitleClick: this.onTitleClick,
+  const isSelectedItem = (item: TreeItemProps): boolean => {
+    if (item.selectableParent && item.items) {
+      return isAllGroupChecked(item.items);
+    }
+
+    return selectedItemIds && selectedItemIds.indexOf(item.id) > -1;
   };
 
-  renderContent(accessibility: ReactAccessibilityBehavior): React.ReactElement[] {
-    const { items, renderItemTitle, selectable } = this.props;
+  const contextValue: TreeRenderContextValue = {
```

Comment:
Ouch. This was previously class property and was stable between rerenders, let's use `useMemo()` to make it stable:

```tsx
const contextValue: TreeRenderContextValue = React.useMemo(() => { /* ... */ }, [onFocusParent, onSiblingsExpand, onFocusFirstChild, onTitleClick])
```

It also means that we should make stable all callbacks there via `React.useCallback()`.

Reply:
Done